ApplicationCode/GeoMech/GeoMechDataModel/RigFemPartResultCalculatorPoreCompressibility.cpp

```diff
@@ -210,52 +214,58 @@ RigFemScalarResultFrames*
 
                             // Calculate pore compressibility
                             double poreCompressibility = inf;
                             if ( deltaPorePressure != 0.0 && porosity != 0.0 )
                             {
-                                poreCompressibility =
-                                    -( biotCoefficient * evData[elmNodResIdx] ) / ( deltaPorePressure * porosity );
+                                double deltaEv      = evData[elmNodResIdx] - referenceEvData[elmNodResIdx];
+                                poreCompressibility = -( biotCoefficient * deltaEv ) / ( deltaPorePressure * porosity );
                                 // Guard against divide by zero: second term can be ignored when bulk modulus is zero,
                                 // which can happens when biot coefficient is 1.0
                                 if ( biotCoefficient != 1.0 && porosity != 1.0 )
                                 {
                                     poreCompressibility += ( 1.0 / bulkModulus ) * ( biotCoefficient / porosity - 1.0 );
                                 }
                             }
-                            poreCompressibilityFrameData[elmNodResIdx] = poreCompressibility;
+                            // Convert from 1/Pa to 1/GPa
+                            poreCompressibilityFrameData[elmNodResIdx] = poreCompressibility * 1.0e9;
 
                             double verticalCompressibility      = inf;
                             double verticalCompressibilityRatio = inf;
 
                             if ( biotCoefficient != 0.0 && deltaPorePressure != 0.0 )
                             {
-                                // Calculate vertical compressibility
-                                verticalCompressibility =
-                                    -verticalStrainData[elmNodResIdx] / ( biotCoefficient * deltaPorePressure );
+                                double deltaStrain = verticalStrainData[elmNodResIdx] -
+                                                     referenceVerticalStrainData[elmNodResIdx];
+
+                                // Calculate vertical compressibility (unit: 1/Pa)
+                                verticalCompressibility = -deltaStrain / ( biotCoefficient * deltaPorePressure );
 
                                 // Calculate vertical compressibility ratio
                                 verticalCompressibilityRatio =
                                     ( verticalCompressibility * youngsModuli * ( 1.0 - poissonRatio ) ) /
                                     ( ( 1.0 + poissonRatio ) * ( 1.0 - 2.0 * poissonRatio ) );
                             }
 
-                            verticalCompressibilityFrameData[elmNodResIdx]      = verticalCompressibility;
+                            // Convert from 1/Pa to 1/GPa
+                            verticalCompressibilityFrameData[elmNodResIdx]      = verticalCompressibility * 1.0e9;
```

Comment:
This is not unit conversion, so it might be overkill to move the scaling factor into conversion class. You decide.